### PR TITLE
Bugfix | Web UI server-side storage

### PR DESCRIPTION
**Pull request by stojkovicv**
## Description

This PR enhances session data storage by moving it from the browser cache to server-side storage. It introduces several auxiliary endpoints and tweaks frontend functionalities to allign with the in-memory writing of the Python classes. Minor styling changes are done, as well.
The changes have been tested from two different devices to confirm consistency.

Related issue: #49

### CODE REVIEW

File: management-ui/server.py [hunk 1]
```diff
@@ -94,11 +94,36 @@ async def new_provisioning_session(app_id: Optional[str] = None, asp_id: Optiona
...
+"""
+Endpoint: Remove all provisioning sessions
+HTTP Method: DELETE
+Path: /remove_all_sessions
+Description: This endpoint will remove all provisioning sessions from the memory.
+"""
+@app.delete("/remove_all_sessions")
+async def remove_all_sessions():    
+    session = await get_session(config)
+    session._M1Session__provisioning_sessions.clear()
```
Comment (davidjwbbc) [inline management-ui/server.py]: You should not be accessing protected members of a class from outside the class, please stick to using M1Session public methods to manipulate the session.

Also clearing the list of known provisioning sessions like this is a bad idea (it won't update the persistent data store for one), if you want to remove provisioning sessions then use `session.provisioningSessionDestroy()` on each known provisioning session id (`session.provisioningSessionIds()`).

File: management-ui/server.py [hunk 2]
```diff
@@ -200,26 +225,45 @@ async def get_provisioning_session_details():
...
-        if cert_id is None:
-            raise HTTPException(status_code=400, detail='Failed to create certificate')
+        else:
+            cert_id = await session.createNewCertificate(provisioning_session_id, extra_domain_names=extra_domain_names)
+            if cert_id is None:
+                raise HTTPException(status_code=400, detail='Failed to create certificate')
+            
+        session_data = session._M1Session__provisioning_sessions.get(provisioning_session_id)
+        if session_data is not None:
+            session_data['certificate_id'] = cert_id
```
Comment (davidjwbbc) [inline management-ui/server.py]: The certificate Id will already be in the list of certificate Ids in the provisioning session after the call to `session.createNewCertificate()`, you shouldn't be setting strange dictionary members.

File: management-ui/server.py [hunk 3]
```diff
@@ -200,26 +225,45 @@ async def get_provisioning_session_details():
...
-    try:
-        cert_id = await session.createNewCertificate(provisioning_session_id, extra_domain_names=extra_domain_names)
-        if cert_id is None:
-            raise HTTPException(status_code=400, detail='Failed to create certificate')
+        else:
+            cert_id = await session.createNewCertificate(provisioning_session_id, extra_domain_names=extra_domain_names)
+            if cert_id is None:
+                raise HTTPException(status_code=400, detail='Failed to create certificate')
+            
+        session_data = session._M1Session__provisioning_sessions.get(provisioning_session_id)
```
Comment (davidjwbbc) [inline management-ui/server.py]: Again, there are methods to access and manipulate provisioning sessions in M1Session. There is no need to bypass the class and access member variables directly.

File: management-ui/server.py [hunk 4]
```diff
@@ -200,26 +225,45 @@ async def get_provisioning_session_details():
...
+HTTP Method: GET
+Path: /get_certificate_id/{provisioning_session_id}
+Description: This endpoint will return the certificate ID for a particular provisioning session.
+"""    
+@app.get("/get_certificate_id/{provisioning_session_id}")
+async def get_certificate_id(provisioning_session_id: str):
+    session = await get_session(config)
+    session_data = session._M1Session__provisioning_sessions.get(provisioning_session_id)
+    if session_data and 'certificate_id' in session_data:
+        return {"certificate_id": session_data['certificate_id']}
```
Comment (davidjwbbc) [inline management-ui/server.py]: You should be using `session.provisioningSessionCertificateIds(provisioning_session_id)` to retrieve a list of certificates, not inventing a single certificate store.

**Review by davidjwbbc — CHANGES_REQUESTED** [review]
There's some changes needed so that these changes respect the class interfaces for M1Session instead of trying to bypass them. Bypassing them could cause misbehaviour in the class as your bypass is not maintaining the internal data integrity maintained by the class itself. It also add the risk that future changes to the internals of the M1Session class break your code when internal data structures are changed.

File: management-ui/server.py (hunk 4 above)
Comment (stojkovicv) [reply management-ui/server.py]: There's been an unclear issue, where I receive empty list of a certificate Ids when using `session.provisioningSessionCertificateIds(provisioning_session_id)`, even though the Certificate is created. 

Another possibility is to fetch this ID directly from the AF response for provisioning session:
```
    "serverCertificateIds": [
        "f9a4cf16-58dc-41ef-8bfc-d7aaf8f810a4"
    ]
 ```

Comment (stojkovicv) [reply management-ui/server.py]: In other words, `session.provisioningSessionCertificateIds` might have memory issue at some point along the way, resulting always with the empty list.
Instead I call `session.certificateIds()`

Comment (davidjwbbc) [reply management-ui/server.py]: That's a bug that should be fixed then. If something is not working as expected then feel free to raise an issue.

**Review by davidjwbbc — APPROVED** [review]

**Pull request merged**